src/main/kotlin/org/jetbrains/intellij/tasks/RunPerformanceTestTask.kt

```diff
@@ -0,0 +1,111 @@
+package org.jetbrains.intellij.tasks
+
+import com.jetbrains.plugin.structure.base.utils.createDir
+import com.jetbrains.plugin.structure.base.utils.exists
+import org.gradle.api.Incubating
+import org.gradle.api.tasks.*
+import org.gradle.kotlin.dsl.property
+import org.jetbrains.intellij.*
+import org.jetbrains.intellij.model.ProfilerName
+import org.jetbrains.intellij.performanceTest.IdeaLogParser
+import org.jetbrains.intellij.performanceTest.PerfTestResult
+import org.jetbrains.intellij.performanceTest.SimpleIJPerfParser
+import org.jetbrains.intellij.performanceTest.TestExecutionFailException
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.absolutePathString
+
+@Incubating
+open class RunPerformanceTestTask : RunIdeTask() {
+
+    private val context = logCategory()
+
+    @get:Input
+    val testDataDir = objectFactory.property<String>()
```

Comment:
please add code documentation to all user-settable properties, so they'll be available in the IDE when editing gradle script